com.unity.ml-agents/Runtime/Communicator/GrpcExtensions.cs

```diff
@@ -274,6 +287,7 @@ public static ObservationProto GetObservationProto(this ISensor sensor, Observat
             }
             else
             {
+                var compressibleSensor = sensor as ICompressibleSensor;
```

Comment:
As mentioned above, you cannot assume that `sensor` is an `ICompressibleSensor`.

Also, the proto code will throw if you pass it a null pointer, so you need to check that before setting.